resalloc_ibm_cloud/powervs/powervs_vm.py

```diff
@@ -0,0 +1,214 @@
+
+        for instance in instances:
+            if instance["serverName"] == instance_name:
+                instance_id = instance["id"]
+                break
+
+        if not instance_id:
+            logger.warning("No instance found with name %s", instance_name)
+            return
+
+        self.client.delete_instance(
+            self.cloud_instance_id, instance_id, delete_data_volumes=True
+        )
+        logger.info(
+            "PowerVS instance %s (ID: %s) deletion initiated",
+            instance_name,
+            instance_id,
+        )
+
+    def _parse_volumes(self, volumes_list: list[str]) -> list[dict]:
+        """
+        Parse volume specifications from a list of strings.
+
+        Each string should be in the format "name:size[:diskType]".
+        """
+        result = []
+        for vol_spec in volumes_list:
+            parts = vol_spec.split(":")
+            if len(parts) < 2:
+                logger.warning("Invalid volume specification: %s", vol_spec)
+                continue
+
+            volume = {
+                "name": parts[0],
+                "size": float(parts[1]),
+            }
+
+            if len(parts) > 2:
+                volume["diskType"] = parts[2]
+
+            result.append(volume)
+
+        return result
+
+    def _wait_for_instance_active(
+        self, instance_id: str, timeout: int = 1200, interval: int = 10
+    ) -> dict:
+        start_time = time.time()
+        while True:
+            if time.time() - start_time > timeout:
+                raise TimeoutError(
+                    f"Instance did not become active within {timeout} seconds"
+                )
+
+            instance = self.client.get_instance(self.cloud_instance_id, instance_id)
+            status = instance.get("status")
+
+            logger.info("Instance status: %s", status)
+
+            if status == "ACTIVE":
+                return instance
+            elif status in ["ERROR", "FAILED"]:
+                raise RuntimeError(f"Instance creation failed with status: {status}")
+
+            sleep(interval)
+
+
+def main() -> int:
+    """
+    Main entry point for PowerVS VM management.
+
+    Returns:
+        Exit code
+    """
+    opts = powervs_arg_parser().parse_args()
+
+    setup_logging(opts.log_level)
+
+    try:
+        credentials = get_powervs_credentials(opts.token_file, opts.region)
+        client = PowerVSClient(credentials)
+        vm_manager = PowerVSVMManager(client, opts.cloud_instance_id)
+
+        if opts.subparser == "create":
+            ip_address = vm_manager.create_vm(opts.instance_name, opts)
+            print(ip_address)
+        elif opts.subparser == "delete":
+            vm_manager.delete_vm(opts.instance_name)
+        else:
+            logger.error("Unknown subcommand: %s", opts.subparser)
+            return 1
+
+        return 0
+
+    except Exception as e:
+        logger.exception("Error in PowerVS VM management: %s", str(e))
+        return 1
```

Comment:
Can you actually return status codes like this in Python instead of doing `sys.exit(1)`? That would be a TIL.